sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1227,6 +1227,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val HIVE_METASTORE_DROP_PARTITION_BY_NAME =
+    buildConf("spark.sql.hive.dropPartitionByName.enabled")
+      .doc("When true, Spark will get partition name rather than partition object " +
+           "to drop partition, which can improve the performance of drop partition.")
+      .version("3.4.0")
```

Comment:
Hi, @sunchao . You need to backport this to branch-3.4.

Reply:
You can do that if you need this. Otherwise, we need to change this to `3.5.0`.

Reply:
thanks for pointing out @dongjoon-hyun ! yes, let me backport this to 3.4.0 too and update the JIRA accordingly.

Reply:
I think it's pretty safe to backport to branch-3.4 since the feature is turned off by default.

Reply:
Thank you for the decision. I also support your decision. Here is my +1.